### PR TITLE
feat: Add SplitPane component to UI Kit [WEB-1682]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
Adds the `SplitPane` component to the UIKit. Since there is only one instance of the component in Determined the API for the component is based on its current usage. 

In Determined we currently specify both `isMobile` and `open` properties for the `SplitPane` however the properties serve to either hide the `left` or `right` pane. 

The component API has been simplified so that the component expects both a `left` and `right` pane property, and an additional prop to hide either the `left` or `right` pane. 

[WEB-1682](https://hpe-aiatscale.atlassian.net/browse/WEB-1682)

[WEB-1682]: https://hpe-aiatscale.atlassian.net/browse/WEB-1682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ClaireNeveu — APPROVED** [review]

**Review by johnkim-det — COMMENTED** [review]
I'm wondering if it makes sense to simplify the API further and not even use `hidePane`. and instead, use conditionals for `left` and `right` props, and render based on that, instead of always rendering both and applying `display: none` conditionally. I feel like it would 1) be easier to write conditionals for `left` and `right` than `hidePane` for showing/hiding right pane at wider screen and toggling between panes at mobile resolution and 2) still have the flexibility of moving away from `open`/`isMobile`

@EmilyBonar thoughts?

File: src/kit/SplitPane.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      setIsDragging(false);
+      throttledOnChange?.cancel();
+      onChange?.(width);
+    };
+    document.addEventListener('mouseup', handleDragStop);
+
+    return () => document.removeEventListener('mouseup', handleDragStop);
+  }, [width, isDragging, onChange, throttledOnChange]);
+
+  const hideHandle = hidePane === Pane.Left || hidePane === Pane.Right;
```
Comment (EmilyBonar) [inline src/kit/SplitPane.tsx:105]: nit: this can just be `hidePane !== undefined"

File: src/kit/SplitPane.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  return (
+    <div className={classnames.join(' ')} ref={refCallback}>
+      <div style={{ display: leftPaneDisplay, width: leftPaneWidth }}>{leftPane}</div>
+      {
+        <div
+          className={css.handle}
+          ref={handle}
+          style={{ display: hideHandle ? 'none' : 'initial' }}
+        />
+      }
```
Comment (EmilyBonar) [inline src/kit/SplitPane.tsx:125]: nit: don't think this needs the brackets

File: src/kit/SplitPane.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  const hideHandle = hidePane === Pane.Left || hidePane === Pane.Right;
+
+  const classnames = [css.base, themeClass];
+
+  if (hidePane !== Pane.Right) {
+    classnames.push(css.showRightPane);
+  }
+
+  const leftPaneWidth = hidePane !== Pane.Right ? width : '100%';
+  const leftPaneDisplay = hidePane === Pane.Left ? 'none' : 'initial';
```
Comment (EmilyBonar) [inline src/kit/SplitPane.tsx:114]: nit: maybe just combine these into a `leftPaneStyle` object? or at least use the same conditional for both.

File: src/kit/SplitPane.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+interface PaneWidths {
+  [Pane.Left]: number;
+  [Pane.Right]: number;
+}
+
+interface Props {
+  leftPane: React.ReactElement;
+  rightPane: React.ReactElement;
+  initialWidth?: number;
+  minimumWidths?: PaneWidths;
```
Comment (EmilyBonar) [inline src/kit/SplitPane.tsx:26]: nit: demonstrate changing minimum widths on the kit page

**Review by EmilyBonar — APPROVED** [review]
Looks good, just a couple tweaks!

**Pull request merged**